pkg/sdk/renderer/argo/dedicated_renderer.go

```diff
@@ -955,16 +997,33 @@ func (r *dedicatedRenderer) addOutputTypeInstancesToGraph(step *WorkflowStep, pr
 		artifactName := r.addTypeInstanceName(name)
 		artifactNamesMap[item.TypeInstanceName] = artifactName
 
+		// select backend
+		upperLayerStepOutputBackendAlias := ptr.StringPtrToString(mappings[name].Backend)
+		backendAlias, err := r.selectBackendAlias(upperLayerStepOutputBackendAlias, stepOutputBackendAlias)
```

Comment:
If alias may be empty, maybe it would be better to pass it as a pointer which can be nil? 🤔  Up to you, but it would be more intuitive from developer's point of view (in my opinion of course)